src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,174 @@
+
+        // text of data from variable 'dataTo'
+        int seventhIndex;
+        int eighthIndex;
+        int ninthIndex;
+        seventhIndex = dateFrom.indexOf(" ");
+        eighthIndex = dateFrom.indexOf(" ", seventhIndex + 1);
+        ninthIndex = dateFrom.indexOf(" ", eighthIndex + 1);
+        String data2;
+        data2 = dateTo.substring(ninthIndex,
+                dateTo.length() - 1); // text of data from variable 'dataTo'
+
+        // find indexes of '.' and ' ' in order to find names, day, year and month from array 'data'
+        int tenthIndex;
+        int eleventhIndex;
+        int twelfthIndex;
+        tenthIndex = data[0].indexOf(".");
+        eleventhIndex = data[0].indexOf(".", tenthIndex + 1);
+        twelfthIndex = data[0].indexOf(" ");
+
+        /* here I check if 'day' is 03, I can not save this amount to int and I save only 3,
+         if 12 or 23 I save this like that
+         */
+        int day3;
+        day3 = Integer.parseInt(dateFrom.substring(sixthIndex + 2, sixthIndex + 3));
+        if (day3 == 0) {
+            day5 = Integer.parseInt(dateFrom.substring(sixthIndex + 3, sixthIndex + 4)); //dayFrom
+        } else {
+            day5 = Integer.parseInt(dateFrom.substring(sixthIndex + 2, sixthIndex + 4)); //dayFrom
+        }
+
+        //the same about month
+        int month3;
+        month3 = Integer.parseInt(dateFrom.substring(sixthIndex + 5, sixthIndex + 6));
+        if (month3 == 0) {
+            month5 = Integer.parseInt(dateFrom.substring(sixthIndex + 6,
+                    sixthIndex + 7));//monthFrom
+        } else {
+            month5 = Integer.parseInt(dateFrom.substring(sixthIndex + 5,
+                    sixthIndex + 7));//monthFrom
+        }
+
+        //here I find data to output
+        int day4;
+        int year3;
+        year3 = Integer.parseInt(dateFrom.substring(sixthIndex + 8, sixthIndex + 12)); //yearFrom
+        day4 = Integer.parseInt(dateTo.substring(ninthIndex, ninthIndex + 1));
+
+        //the same about day from String 'dateTo'
+        if (day4 == 0) {
+            day6 = Integer.parseInt(dateTo.substring(ninthIndex + 1, ninthIndex + 2));//dayTo
+        } else {
+            day6 = Integer.parseInt(dateTo.substring(ninthIndex, ninthIndex + 2));//dayTo
+        }
+
+        //the same about month from String 'dateTo'
+        int month4;
+        month4 = Integer.parseInt(dateTo.substring(ninthIndex + 3, ninthIndex + 4));
+        if (month4 == 0) {
+            month6 = Integer.parseInt(dateTo.substring(ninthIndex + 4, ninthIndex + 5));//monthTo
+        } else {
+            month6 = Integer.parseInt(dateTo.substring(ninthIndex + 3, ninthIndex + 5)); //monthTo
+        }
+        //the same about year from String 'dateTo'
+        int year4;
+        StringBuilder result = new StringBuilder();
+        year4 = Integer.parseInt(dateTo.substring(ninthIndex + 6, ninthIndex + 10)); //yearTo
```

Comment:
Magic numbers such as '6', '10' etc., used for parsing dates should be declared as constants with descriptive names.